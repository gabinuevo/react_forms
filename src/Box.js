import React, { Component } from 'react';


/**component displays div w/: 
 * props: {
 *      background color, 
 *      width 
 *      height 
 * }
 */
class Box extends Component {
    render() {
        let style = {
        height: this.props.height,
        width: this.props.width,
        backgroundColor: this.props.backgroundColor
        }
        return (<div className = "Box">
                    <div style = {style}></div>
                    <button onClick={this.props.triggerRemove}>X</button>
                </div>)
    }
}

export default Box