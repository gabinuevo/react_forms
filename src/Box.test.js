import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Box from "./Box";

let wrapper;

it("renders succesfully", function() {
    shallow(<Box />);
});

it("matches snapshot on initial render", function() {
    wrapper = shallow(<Box />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });