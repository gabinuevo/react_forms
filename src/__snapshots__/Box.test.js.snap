// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on initial render 1`] = `
<div
  className="Box"
>
  <div
    style={
      Object {
        "backgroundColor": undefined,
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <button>
    X
  </button>
</div>
`;
